lib/app.rb

```diff
@@ -0,0 +1,63 @@
+require_relative 'constants'
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'stdio'
+require_relative 'rental'
+
+class App
+  include Constants
+  def initialize
+    @cases = {
+      '1': -> { StandardIO.list_books(book: Book) },
+      '2': -> { StandardIO.list_people(Student.all + Teacher.all) },
+      '3': -> { create_person },
+      '4': -> { create_book },
+      '5': -> { create_rental },
+      '6': -> { StandardIO.list_rental(Student.all + Teacher.all) },
+      '7': -> { puts 'Thank you for using the app' }
+    }
+
+    @input_error = 'Please input a number between 1-7'
+  end
+
+  def run
+    input = nil
+    until input == '7'
+      PROMPTS.each { |prompt| puts prompt }
+      input = gets.chomp
+      if valid_input?(input)
+        call_service(input)
+      else
+        puts @input_error
+      end
+    end
+  end
```

Comment:
- [ ] You should follow the single responsibility principle. All requests for input and input validation should've been placed in the `main.rb` file. Please [check this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid
) article if you need a refresher.